src/daemon.rs

```diff
@@ -378,19 +379,24 @@ impl Daemon {
 
     fn handle_request_batch(&self, method: &str, params_list: &[Value]) -> Result<Vec<Value>> {
         let id = self.message_id.next();
-        let reqs = params_list
+        let chunks = params_list
             .iter()
             .map(|params| json!({"method": method, "params": params, "id": id}))
-            .collect();
+            .chunks(50_000); // Max Amount of batched requests
         let mut results = vec![];
```

Comment:
nit: consider preallocating `results`:
```suggestion
        let mut results = Vec::with_capacity(params_list.len());
```

Reply:
iirc for Ts larger than 1 byte, but less than 1kB, the starting capacity is 4 and doubles each time it reallocates.
Meaning we can save 14 reallocations for a size of 65536 items. 17 reallocations for 524288 items.

Considering the speed of reallocation vs. the speed of the round trip to RPC and back for large batches, the performance cost for all those reallocations is probably not that big of a deal in comparison.

That said, the change to the code is minimal and there's no reason NOT to do it.